### ISSUE TITLE
Vending Machine Flow Only Sees One Card

### ISSUE THREAD

**Issue by Kurfursten** [d3athrow/vgstation13#21064]
Annoying for traders, even if you have cash on your card it will buzz at you for having no associated bank account so you have to take out money and pay by hand

**Comment by SonixApache** [comment]
you arguably caused this yourself with #19762

**Comment by Kurfursten** [comment]
I don't think you understand, this isn't about having a connection to the accounts database. Even if the database is active, you can't use a vending machine with no bank account on your ID.

**Comment by Kurfursten** [comment]
Can't reproduce now. Weird.

**Comment by Kurfursten** [comment]
Figured it out. If you are holding (or wearing it, like in your pocket) a debit card (e.g.: the trader debit card) the vending machine will never check for your ID, always prioritizing the debit card.

It would be nice if it would look for another card after the first.
